source/distributed/shared_tria.cc

```diff
@@ -183,11 +195,8 @@ namespace parallel
           if (cell->is_locally_owned())
             n_my_cells += 1;
 
-        unsigned int total_cells;
-        int ierr = MPI_Allreduce (&n_my_cells, &total_cells, 1,
-                                  MPI_UNSIGNED, MPI_SUM, this->get_communicator());
-        AssertThrowMPI(ierr);
-
+        const unsigned int total_cells
+          = Utilities::MPI::sum(n_my_cells, this->get_communicator());
```

Comment:
nice cleanup!

Reply:
It seems a bit risky to me to use an `unsigned int` here for the global number of cells. Would you mind switching to `std::size_t`?

Reply:
It's a shared triangulation, so the sum of the number of locally owned cells (`n_my_cells`) adds up to the number of active cells each processor stores itself locally. I think it's unlikely that we're ever going to have machines where a single MPI process will store more than 4B cells :-)

Reply:
Note that the type of ``n_active_cells`` is an ``unsigned int`` and is equal to ``total_cells``.

Reply:
good point; I forgot that all cells are stored locally in this case.